src/web/components/StickyBottomBanner/BrazeBanner.tsx

```diff
@@ -100,11 +116,18 @@ const getMessageFromBraze = async (
             };
 
             if (extras) {
+                const appboyTimeTaken = appboyTiming.end()
```

Comment:
Nice! Yeah I think this will more accurately measure the time taken from braze initialising to getting a message. One small improvement we could make is that in _theory_ this callback could fire multiple times, though we only care about the first one. The way we handle that in terms of what `canShow` returns is through the promise we return in that a promise is guaranteed to only resolve once. So if this callback did fire multiple times, only the first call would have any effect outwardly (i.e. the first time we call `resolve` in here). So it feels to me like the most correct thing would be to handle the timing end when the promise resolves e.g.:

So instead of:

```js
return new Promise((resolve) => {
  // ...
})
```

We could do:
```js
const canShowPromise = new Promise((resolve) => {
  // ...
});

canShowPromise.then((result) => {
  const appboyTimeTaken = appboyTiming.end();

  record({
    component: 'braze-appboy-timing',
    value: appboyTimeTaken,
  });
}).catch(() => {
  console.log("Something went wrong recording timing");
});

return canShowPromise;
```

What do you reckon?

Reply:
Ah yes, good point - I hadn't thought about multiple calls. This makes perfect sense to me.